ownCloud/Settings/Passcode/AppLockManager.swift

```diff
@@ -263,4 +273,30 @@ class AppLockManager: NSObject {
 			}
 		}
 	}
+
+    // MARK: - Biometrical Unlock
+
+    func authenticateWithBiometrical() {
+
+        if  shouldDisplayLockscreen, biometricalSecurityEnabled {
+
+            let context = LAContext()
+
+            // Check if the device can evaluate the policy.
+            if context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: nil) {
+                context.evaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, localizedReason: "Unlock passcode".localized) { (success, _) in
+                    if success {
```

Comment:
This codes lacks error handling. Especially for `LAError.code.biometryLockout`, which should be counted against failed attempts via `attemptUnlock(nil)`.